hybig/color_utility.py

```diff
@@ -87,7 +90,16 @@ def get_color_palette(
         return get_remote_palette_from_source(source)
     except HyBIGNoColorInformation:
         try:
-            return convert_colormap_to_palette(dataset.colormap(1))
+            ds_cmap = dataset.colormap(1)
+            # very defensive since this function is not documented in rasterio
+            ndv_tuple: tuple[float, ...] = dataset.get_nodatavals()
+            if ndv_tuple is not None and len(ndv_tuple) > 0:
+                # this service only supports one ndv, so just use the first one
+                # (usually the only one)
+                ds_cmap['nv'] = ds_cmap[ndv_tuple[0]]
+                # then remove the value associated with the ndv key
+                ds_cmap.pop(ndv_tuple[0])
+            return convert_colormap_to_palette(ds_cmap)
```

Comment:
This looks like you know more about where color tables and nodata values might be, but if there was a test for it, it might be easier for me to understand.